pkg/schedule/checker/rule_checker.go

```diff
@@ -502,15 +512,15 @@ loopFits:
 				continue
 			}
 			// make sure the orphan peer is healthy.
-			if isUnhealthyPeer(orphanPeer.GetId()) {
+			if isUnhealthyPeer(orphanPeer.GetId()) || isDisconnectedPeer(orphanPeer) {
 				continue
 			}
 			// no consider witness in this path.
 			if pinDownPeer.GetIsWitness() || orphanPeer.GetIsWitness() {
 				continue
 			}
-			// down peer's store should be down.
-			if !c.isStoreDownTimeHitMaxDownTime(pinDownPeer.GetStoreId()) {
+			// down peer's store should be disconnected
+			if !isDisconnectedPeer(pinDownPeer) {
```

Comment:
use `isDisconnectedPeer` rather than `isStoreDownTimeHitMaxDownTime`, because we use stricter check in `isDisconnectedPeer`